lib/Platform/Intl/java/com/facebook/hermes/intl/DateTimeFormat.java

```diff
@@ -244,10 +255,10 @@ private void initializeDateTimeFormat(List<String> locales, Map<String, Object>
     if (JSObjects.isUndefined(timeZone)) {
       timeZone = DefaultTimeZone();
     } else {
-      String normalizedTimeZone = normalizeTimeZoneName(JSObjects.getJavaString(timeZone));
-      if (!isValidTimeZoneName(normalizedTimeZone)) {
+      if (!isValidTimeZoneName(timeZone.toString())) {
         throw new JSRangeErrorException("Invalid timezone name!");
       }
```

Comment:
As mentioned, we don't need to perform this check any more, since the normalising function will automatically reject anything that isn't a valid timezone anyway.